### PR TITLE
Make 'select' type Input trigger onChange and update state correctly.

### PR DESCRIPTION
**Pull request by jareth**
Adding onChange to React select components doesn't work with Materialize's `<ul>` based virtual styled select component. See: https://github.com/Dogfalo/materialize/issues/1160

### CODE REVIEW

File: src/Input.js [hunk 1]
```diff
@@ -14,6 +14,10 @@ class Input extends React.Component {
   componentDidMount() {
     if (this.props.type === 'select' && !this.props.browserDefault && typeof $ !== 'undefined') {
       $(this.refs.inputEl).material_select();
+      $(this.refs.inputEl).on('change', this._onChange);
+      if (!!this.props.onChange) {
```
Comment (isaiah) [inline src/Input.js]: You should move this to the `this._onChange` so that the existing function is not override.

**Comment by isaiah** [comment]
I've fixed the lint, please rebase.

**Comment by jareth** [comment]
- props.onChange call moved into this._onChange
- rebased

I noticed some of the other form components onChange aren't working as I would expect during testing this, but I would deal with them as a separate issue.

**Comment by isaiah** [comment]
Thanks.

**Pull request merged**